pkg/sources/git/git.go

```diff
@@ -174,111 +177,98 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ .
 }
 
 // scanRepos scans the configured repositories in s.conn.Repositories.
-func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk) error {
+func (s *Source) scanRepos(ctx context.Context, reporter sources.ChunkReporter) error {
 	if len(s.conn.Repositories) == 0 {
 		return nil
 	}
 	totalRepos := len(s.conn.Repositories) + len(s.conn.Directories)
-	// TODO: refactor to remove duplicate code
+	for i, repoURI := range s.conn.Repositories {
+		s.SetProgressComplete(i, totalRepos, fmt.Sprintf("Repo: %s", repoURI), "")
+		if len(repoURI) == 0 {
+			continue
+		}
+		if err := s.scanRepo(ctx, repoURI, reporter); err != nil {
+			ctx.Logger().Info("error scanning repository", "repo", repoURI, "error", err)
+			continue
```

Comment:
is this necessary or is it just here for symmetry?

Reply:
Not necessary - it's similar to the `return` I did at the end of a function. No functional difference, but marginally better if we add code after it.